test/distributed/test_c10d.py

```diff
@@ -340,6 +340,82 @@ def _test_numkeys_delkeys(self, fs):
         self.assertEqual(b"value1", fs.get("key1"))
         self.assertEqual(b"value2", fs.get("key4"))
 
+        store = self._create_store()
+        keys = [str(i) for i in range(10)]
+        t0 = time.perf_counter()
+        [store.set(k, k) for k in keys]
+        dur_set = time.perf_counter()
+        t0 = time.perf_counter()
+        [store.delete_key(k) for k in keys]
+        dur_delete = time.perf_counter()
+        print(dur_set, dur_delete)
+
+    def _test_delkey_perf_worker(self, index, addr, port, world_size, messages):
+        try:
+            store = dist.TCPStore(addr, port, world_size, timeout=timedelta(seconds=10))
+            if index == 0:
+                keys = [str(i) for i in range(1, world_size)]
+
+                for i in range(5):
+                    start = time.perf_counter()
+                    for k in keys:
+                        store.wait([k])
+                    dur_wait = (time.perf_counter() - start) * 1000
+
+                    start = time.perf_counter()
+                    for k in keys:
+                        store.get(k)
+                    dur_store = (time.perf_counter() - start) * 1000
+
+                    start = time.perf_counter()
+                    for k in keys:
+                        store.delete_key(k)
+                    dur_delete = (time.perf_counter() - start) * 1000
+
+                    if dur_delete > 5 * max(dur_wait, dur_store):
```

Comment:
This seems quite prone to flakiness, at the least could we make it a much more generous threshold?